src/Compilers/Core/Portable/Text/StringBuilderText.cs

```diff
@@ -89,5 +91,45 @@ public override void CopyTo(int sourceIndex, char[] destination, int destination
         {
             _builder.CopyTo(sourceIndex, destination, destinationIndex, count);
         }
+
+        public override void Write(TextWriter textWriter, TextSpan span, CancellationToken cancellationToken = default)
+        {
+#if NETCOREAPP
+            if (span.Start == 0 && span.Length == this.Length)
+            {
+                textWriter.Write(_builder);
+            }
+            else
+            {
+                if (span.Length == 0)
+                    return;
+
+                int chunkOffset = 0;
```

Comment:
doc waht this is.

Reply:
@CyrusNajmabadi  would you be ok with `// Index of the current chunk in the text.`?

Reply:
Or do you mean document the entire loop?

Reply:
basically, i'm not sure what chunkOffset means.  offset is a very vague term that could be interpreted in a lot of ways.

Reply:
I didn't name it index because that would sound like it's the index an offset array.

Reply:
@jaredpar Do you have any suggestions for a better name?